### PR TITLE
Change Legacy Hook with new Hook dispatche event: actionAdminProducts

### PR DESCRIPTION
**Pull request by luigimassa**
Upgrade from Legacy Hook::exec to new Hook dispatcher

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Upgrade from Legacy Hook::exec to new Hook dispatcher
| Type?             |  improvement 
| Category?         | BO
| BC breaks?        |  no
| Deprecations?     |  no
| Fixed ticket?     | Fix 28680
| Related PRs       |
| How to test?      | BO > Product > List product
| Possible impacts? |

### CODE REVIEW

**Comment by kpodemski** [comment]
Hello @luigimassa 

I think it's not a change you wanted to introduce?

**Comment by luigimassa** [comment]
> Hello @luigimassa
> 
> I think it's not a change you wanted to introduce?

now fix

**Review by kpodemski — DISMISSED** [review]

File: src/Adapter/Product/AdminProductDataProvider.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ public function getPersistedFilterParameters()
         $employee = Context::getContext()->employee;
         $employeeId = $employee->id ?: 0;
 
-        $cachedFilters = $this->cache->getItem("app.product_filters_${employeeId}");
+        $cachedFilters = $this->cache->getItem("app.product_filters_{$employeeId}");
```
Comment (matks) [inline src/Adapter/Product/AdminProductDataProvider.php]: Can you explain why you do this change? I don't see the intent behind it

File: src/Adapter/Product/AdminProductDataProvider.php [hunk 2]
```diff
@@ -165,11 +165,11 @@ public function persistFilterParameters(array $parameters)
 
         $this->entityManager->flush();
 
-        //Flush cache
+        // Flush cache
         $employee = Context::getContext()->employee;
         $employeeId = $employee->id ?: 0;
 
-        $this->cache->deleteItem("app.product_filters_${employeeId}");
+        $this->cache->deleteItem("app.product_filters_{$employeeId}");
```
Comment (matks) [inline src/Adapter/Product/AdminProductDataProvider.php:180]: Can you explain why you do this change? I don't see the intent behind it

File: src/Adapter/Product/AdminProductDataProvider.php [hunk 3]
```diff
@@ -338,7 +338,7 @@ public function getCatalogProductList(
         $sqlWhere[] = 'state = ' . Product::STATE_SAVED;
 
         // exec legacy hook but with different parameters (retro-compat < 1.7 is broken here)
-        Hook::exec('actionAdminProductsListingFieldsModifier', [
+        $this->hookDispatcher->dispatchWithParameters('actionAdminProductsListingFieldsModifier', [
```
Comment (matks) [inline src/Adapter/Product/AdminProductDataProvider.php:349]: According to PHPStan you are using an undefined property

**Review by matks — CHANGES_REQUESTED** [review]
Dear @luigimassa I apologize for this very late code review. Please see my comment: I need informations and CI fix for this PR

File: src/Adapter/Product/AdminProductDataProvider.php (hunk 3 above)
Comment (luigimassa) [reply src/Adapter/Product/AdminProductDataProvider.php:349]: fixed.

**Comment by kpodemski** [comment]
Hello @luigimassa 

If you reply to us after making proper changes, it is easier to track this as the label "Waiting for author" will be removed.

I didn't notice you sent changes. Could you please rebase your PR so we can quickly review it?

Thank you for contributing!

**Comment by kpodemski** [comment]
Hi @luigimassa 

Since we have had no news from you for more than 20 days, I'm closing this PR. Feel free to re-open it if you find and motivation to finish it. Don't hesitate to ping me if you have any questions or doubts.

**Pull request closed without merging**